### PR TITLE
Updated update-method module to JUnit 5

### PR DESCRIPTION
**Pull request by charlesfinley**
Updated update-method module to JUnit 5

> For detailed contributing instructions see https://github.com/iluwatar/java-design-patterns/wiki/01.-How-to-contribute

### CODE REVIEW

**Comment by ohbus** [comment]
@charlesfinley This looks like a duplicate of #1541 
Please clarify this and submit a single Pull Request.

**Comment by charlesfinley** [comment]
> @charlesfinley This looks like a duplicate of #1541
> Please clarify this and submit a single Pull Request.

This PR includes the changes from 1541 and adds more. #1541  can be closed in favor of this PR if necessary.

**Review by ohbus — APPROVED** [review]
Please try to provide a detailed documentation in the comment section of this Pull Request of all the changes you have made to that It becomes easier for other reviewers.

Overall a Good Pull Request.
Thanks.

Wait for @iluwatar to make the final commit.

File: trampoline/src/main/java/com/iluwatar/trampoline/Trampoline.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ public T get() {
         return trampoline(this);
       }
 
-      T trampoline(final Trampoline<T> trampoline) {
+      private T trampoline(final Trampoline<T> trampoline) {
```
Comment (iluwatar) [inline trampoline/src/main/java/com/iluwatar/trampoline/Trampoline.java:101]: Could you please describe this change a bit, why is it needed? This particular snippet is also part of the pattern's `README.md` (which is rendered on the https://java-design-patterns.com web site) so all the changes should be reflected there.

**Review by iluwatar — CHANGES_REQUESTED** [review]

**Comment by iluwatar** [comment]
@charlesfinley thanks for the pull request! The JUnit 5 related changes look ok, but see my comment about the change to Trampoline pattern.

File: trampoline/src/main/java/com/iluwatar/trampoline/Trampoline.java (hunk 1 above)
Comment (ohbus) [reply trampoline/src/main/java/com/iluwatar/trampoline/Trampoline.java]: @charlesfinley can you please clarify on this part as why we modified this access specifier to `private`

We would love to merge the code to master.

Comment (charlesfinley) [reply trampoline/src/main/java/com/iluwatar/trampoline/Trampoline.java]: Will update soon with explanation. Sorry for the delay.

Comment (charlesfinley) [reply trampoline/src/main/java/com/iluwatar/trampoline/Trampoline.java]: @iluwatar @ohbus - Reverted the change. It's no longer required.

**Review by ohbus — CHANGES_REQUESTED** [review]
LGTM

Thank you so much @charlesfinley for this contribution.

Can please you look at the static issues reported by sonarbot and kindly make the updates to fix it.

**Comment by charlesfinley** [comment]
> LGTM
> 
> Thank you so much @charlesfinley for this contribution.
> 
> Can please you look at the static issues reported by sonarbot and kindly make the updates to fix it.

I can't seem to find the sonarbot issues - can you link me? I'm seeing all 3 checks passing.

**Comment by ohbus** [comment]
The 28 code smells that seems to be reported by the sonarbot comment.

**Comment by charlesfinley** [comment]
> The 28 code smells that seems to be reported by the sonarbot comment.
> 
> Here they are https://sonarcloud.io/project/issues?id=iluwatar_java-design-patterns&pullRequest=1542&resolved=false&types=CODE_SMELL

Looks like all of these pre-date my changes. I'll clean them up and push up a commit shortly.

**Comment by charlesfinley** [comment]
@ohbus Code smells resolved.

File: saga/src/test/java/com/iluwatar/saga/choreography/SagaApplicationTest.java [hunk 2]
```diff
@@ -32,9 +32,9 @@
...
  * empty test
  */
-public class SagaApplicationTest {
+class SagaApplicationTest {
   @Test
-  public void shouldExecuteWithoutException() {
+  void shouldExecuteWithoutException() {
     assertDoesNotThrow(() -> SagaApplication.main(new String[]{}));
   }
 }
```
Comment (ohbus) [inline saga/src/test/java/com/iluwatar/saga/choreography/SagaApplicationTest.java:40]: Please add an extra line here

File: saga/src/test/java/com/iluwatar/saga/orchestration/SagaApplicationTest.java [hunk 3]
```diff
@@ -28,10 +28,10 @@
...
  */
-public class SagaApplicationTest {
+class SagaApplicationTest {
 
   @Test
-  public void mainTest() {
+  void mainTest() {
     SagaApplication.main(new String[]{});
   }
 }
```
Comment (ohbus) [inline saga/src/test/java/com/iluwatar/saga/orchestration/SagaApplicationTest.java:37]: add an empty line here

File: update-method/src/test/java/com/iluwatar/updatemethod/WorldTest.java [hunk 4]
```diff
@@ -27,9 +27,9 @@
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (ohbus) [inline update-method/src/test/java/com/iluwatar/updatemethod/WorldTest.java:30]:
```suggestion
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertFalse;
```

**Review by ohbus — CHANGES_REQUESTED** [review]
These are the final set of changes that would make this PR great!

Please look into this and we can merge this after!

Thank you so much for your changes @charlesfinley

File: update-method/src/test/java/com/iluwatar/updatemethod/WorldTest.java (hunk 4 above)
Comment (charlesfinley) [reply update-method/src/test/java/com/iluwatar/updatemethod/WorldTest.java]: Fixed the star import.

File: saga/src/test/java/com/iluwatar/saga/choreography/SagaApplicationTest.java (hunk 2 above)
Comment (charlesfinley) [reply saga/src/test/java/com/iluwatar/saga/choreography/SagaApplicationTest.java:41]: You want an empty line after the method declaration and before the assertion?

Comment (ohbus) [reply saga/src/test/java/com/iluwatar/saga/choreography/SagaApplicationTest.java:41]: Yes an empty line at the end of the file.

Comment (charlesfinley) [reply saga/src/test/java/com/iluwatar/saga/choreography/SagaApplicationTest.java]: Ah, duh. I need coffee. Updated!

File: saga/src/test/java/com/iluwatar/saga/orchestration/SagaApplicationTest.java (hunk 3 above)
Comment (charlesfinley) [reply saga/src/test/java/com/iluwatar/saga/orchestration/SagaApplicationTest.java]: Done

**Review by ohbus — APPROVED** [review]
LGTM

Thank you @charlesfinley

**Pull request merged**